src/gui_common/DraggableScrollContainer.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using Godot;
+
+/// <summary>
+///   A scroll container which can be also moved by clicking and dragging.
+/// </summary>
+public class DraggableScrollContainer : ScrollContainer
+{
+    [Export]
+    public NodePath? ContentPath;
+
+    /// <summary>
+    ///   Whether we're currently scrolling, to prevent dragging.
+    /// </summary>
+    private bool scrolling;
+
+    /// <summary>
+    ///   Whether we're currently dragging, to prevent us starting dragging again.
+    /// </summary>
+    private bool dragging;
+
+    /// <summary>
+    ///   Timer used for scroll bug workaround in Godot (https://github.com/godotengine/godot/issues/22936).
+    /// </summary>
+    private float scrollingTimer;
+
+    private bool showScrollbars = true;
+
+    private Control content = null!;
+    private Vector2 contentScale = Vector2.One;
+
+    [Export]
+    public float ZoomFactor { get; set; } = 1.05f;
+
+    /// <summary>
+    ///   Controls whether the horizontal and vertical scrollbars should be shown or not. This doesn't enable nor
+    ///   disable the scrollbars.
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     Workaround until scrollbar hiding feature <see href="https://github.com/godotengine/godot/pull/48008"/>
+    ///     is available.
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public bool ShowScrollbars
+    {
+        get => showScrollbars;
+        set
+        {
+            showScrollbars = value;
+            UpdateScrollbars();
+        }
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        if (ContentPath == null)
+            throw new InvalidOperationException("Path to child control is expected");
+
+        content = GetNode<Control>(ContentPath);
+
+        GetVScrollbar().Connect("scrolling", this, nameof(OnScrollStarted));
+        GetHScrollbar().Connect("scrolling", this, nameof(OnScrollStarted));
```

Comment:
Actually it seems like `scrolling` is also never emitted, I put a debug print in `OnScrollStarted` and it never got called.

Reply:
Changed it to use `values_changed` instead.